test/roles.js

```diff
@@ -0,0 +1,942 @@
+      transactions.push(new Transaction(refBlockNumber, fixture.getNextTxId(), 'voter2', 'roles', 'approveCandidate', '{ "id": "1" }'));
+      transactions.push(new Transaction(refBlockNumber, fixture.getNextTxId(), 'voter3', 'roles', 'approveCandidate', '{ "id": "1" }'));
+      transactions.push(new Transaction(refBlockNumber, fixture.getNextTxId(), 'voter4', 'roles', 'approveCandidate', '{ "id": "1" }'));
+      transactions.push(new Transaction(refBlockNumber, fixture.getNextTxId(), 'voter4', 'roles', 'approveCandidate', '{ "id": "2" }'));
+
+      let block = {
+        refHiveBlockNumber: refBlockNumber,
+        refHiveBlockId: 'ABCD1',
+        prevRefHiveBlockId: 'ABCD2',
+        timestamp: '2021-03-12T00:00:00',
+        transactions,
+      };
+
+      await fixture.sendBlock(block);
+      
+      let res = await fixture.database.getLatestBlockInfo();
+      // console.log(res);
+      await tableAsserts.assertNoErrorInLastBlock();
+
+      // weight asserts
+      await assertUserWeight('voter1', 'PRO', '1000.00000000');
+      await assertUserWeight('voter2', 'PRO', '10000.00000000');
+      await assertUserWeight('voter3', 'PRO', '100000.00000000');
+      await assertUserWeight('voter4', 'PRO', '1000000.00000000');
+      await assertUserApproval('voter1', 1);
+      await assertUserApproval('voter2', 1);
+      await assertUserApproval('voter3', 1);
+      await assertUserApproval('voter4', 1);
+      await assertWeightConsistency(1, 'PRO');
+      await assertWeightConsistency(2, 'PRO');
+      await assertWeightConsistency(3, 'PRO');
+      await assertWeightConsistency(4, 'PRO');
+
+
+      refBlockNumber = fixture.getNextRefBlockNumber();
+      transactions = [];
+      transactions.push(new Transaction(refBlockNumber, fixture.getNextTxId(), 'voter4', 'tokens', 'stake', '{ "to": "voter4", "symbol": "PRO", "quantity": "1", "isSignedWithActiveKey": true }'));
+      transactions.push(new Transaction(refBlockNumber, fixture.getNextTxId(), 'voter3', 'tokens', 'unstake', '{ "symbol": "PRO", "quantity": "1", "isSignedWithActiveKey": true }'));
+      transactions.push(new Transaction(refBlockNumber, fixture.getNextTxId(), 'voter3', 'roles', 'setApplyActive', '{ "roleId": "2", "active": false, "isSignedWithActiveKey": true }'));
+
+      block = {
+        refHiveBlockNumber: refBlockNumber,
+        refHiveBlockId: 'ABCD1',
+        prevRefHiveBlockId: 'ABCD2',
+        timestamp: '2021-03-13T00:00:00',
+        transactions,
+      };
+      await fixture.sendBlock(block);
+
+      res = (await fixture.database.getLatestBlockInfo());
+      // console.log(res);
+      await tableAsserts.assertNoErrorInLastBlock();
+
+      await assertWeightConsistency(1, 'PRO');
+      await assertWeightConsistency(2, 'PRO');
+      await assertWeightConsistency(3, 'PRO');
+      await assertWeightConsistency(4, 'PRO');
+
+      assert.ok(res.virtualTransactions.length > 0, 'Expected to find virtualTransactions');
```

Comment:
For this test case, how are the ticks set up to generate these virtual transactions?  Maybe I missed it, but I didn't see any code to configure the per-block ticking behavior.